node/src/test/scala/coop/rchain/node/configuration/hocon/HoconConfigurationSpec.scala

```diff
@@ -2,264 +2,154 @@ package coop.rchain.node.configuration.hocon
 
 import java.nio.file.Paths
 
-import scala.concurrent.duration._
+import com.typesafe.config.ConfigFactory
 
-import coop.rchain.casper.CasperConf
+import scala.concurrent.duration._
+import coop.rchain.casper.{CasperConf, GenesisBlockData, GenesisCeremonyConf, RoundRobinDispatcher}
 import coop.rchain.comm.PeerNode
-import coop.rchain.node.configuration
-
-import com.typesafe.config.ConfigFactory
+import coop.rchain.comm.transport.TlsConf
+import coop.rchain.node.configuration.{
+  ApiServer,
+  Metrics,
+  NodeConf,
+  PeersDiscovery,
+  ProtocolClient,
+  ProtocolServer,
+  Storage
+}
 import org.scalatest.{FunSuite, Matchers}
+import pureconfig.{ConfigReader, ConfigSource, ConvertHelpers}
+import pureconfig.generic.ProductHint
+import pureconfig.generic.auto._
 
 class HoconConfigurationSpec extends FunSuite with Matchers {
 
-  test("Parse server section") {
-    val conf =
-      """
-      |rnode {
-      |  server {
-      |    network-id = "testnet"
-      |    host = 1.2.3.4
-      |    host-dynamic = true
-      |    upnp = false
-      |    port = 40400
-      |    port-http = 40403
-      |    port-kademlia = 40404
-      |    use-random-ports = true
-      |    send-timeout = 2 seconds
-      |    standalone = true
-      |    bootstrap = "rnode://de6eed5d00cf080fc587eeb412cb31a75fd10358@52.119.8.109?protocol=40400&discovery=40404"
-      |    data-dir = /root/.rnode
-      |    store-size = 1G
-      |    dag-storage-size = 512M
-      |    map-size = 1G
-      |    max-connections = 500
-      |    allow-private-addresses = true
-      |    max-message-size = 256K
-      |    max-stream-message-size = 200M
-      |    packet-chunk-size = 64K
-      |    message-consumers = 8
-      |    fault-tolerance-threshold = 0.2
-      |    synchrony-constraint-threshold = 0.3333333333333333
-      |    height-constraint-threshold = 100
-      |    reporting = false
-      |    api-max-blocks-limit = 50
-      |  }
-      |}
-    """.stripMargin
+  test("Parse default config") {
+    val defaultConfig = ConfigSource
+      .resources("defaults.conf")
+      .withFallback(
+        ConfigSource.string(
+          s"default-data-dir = /var/lib/rnode"
+        )
+      )
 
-    val expectedServer =
-      configuration.Server(
+    // Custom reader for PeerNode type is required
+    implicit val PeerNodeReader = ConfigReader.fromString[PeerNode](
+      ConvertHelpers.catchReadError(s => PeerNode.fromAddress(s).getOrElse(null))
+    )
```

Comment:
Using `null` can be avoided here.
```scala
implicit val PeerNodeReader = ConfigReader.fromStringTry[PeerNode](
  ConvertHelpers.catchReadError(PeerNode.fromAddress(_).toTry)
)
```

Reply:
I changed this is other places to
```scala
implicit val PeerNodeReader = ConfigReader.fromString[PeerNode](
      ConvertHelpers.catchReadError(s => PeerNode.fromAddress(s).right.get)
    )
```
are you ok with this option?

Reply:
https://github.com/rchain/rchain/pull/2912/commits/7b1015b6b360d2d044f73a9546b773fb82a1d712

Reply:
I made a comment there https://github.com/rchain/rchain/pull/2912#discussion_r414403759.

Reply:
Fixed